spec/support/realistic_factory.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'realistic factory' do
+  # By 'Realistic' we mean a factory that has all the attributes and relationships
+  # conforming to the schema rather than static data.
+  let(:random_object) { FactoryBot.create("realistic_#{subject.class.to_s.underscore}") }
+
+  it { expect(random_object).to be_valid }
```

Comment:
nice, but is this `be_valid`, not just testing its activerecord model validity. or is that what it is intended. The comment's reference to schema makes me think of json schema?!

Reply:
The intention was to test the model, rather than the schema in this case as we want to be able to create persistent objects.
We have another shared context to match the schema validity:
https://github.com/ministryofjustice/hmcts-common-platform-mock-api/blob/a6bb00f157df90e61d629c832fee51d40f1f2a56/spec/support/shared_contexts.rb#L3-L7